src/App.js

```diff
@@ -9,24 +9,46 @@ import { Box } from "@mui/system";
 
 function App() {
   const [mealsData, setMealsData] = useState(() => {
-    const saved = localStorage.getItem("mealsData");
-    if (saved) {
-      const initialValue = JSON.parse(saved);
+    const savedMealsData = localStorage.getItem("mealsData");
+
+    if (savedMealsData) {
+      const initialValue = JSON.parse(savedMealsData);
       return initialValue;
     } else {
       return meals;
     }
   });
 
+  const [allIngridients, setAllIngridients] = useState(() => {
+    const savedAllIngridients = localStorage.getItem("allIngridients");
+    if (savedAllIngridients) {
+      const initialValue = JSON.parse(savedAllIngridients);
+      return initialValue;
+    } else {
+      return allIngridients;
+    }
+  });
+
   const navigate = useNavigate();
 
   useEffect(() => {
     localStorage.setItem("mealsData", JSON.stringify(mealsData));
-  }, [mealsData]);
+    localStorage.setItem("allIngridients", JSON.stringify(allIngridients));
+  }, [mealsData, allIngridients]);
+
+  const handleNewToAllIngridients = (meal) => {
+    meal.ingridients.forEach((mealIngridient) => {
+      mealIngridient = mealIngridient.toLowerCase();
+      if (!allIngridients.includes(mealIngridient)) {
+        console.log("NEW INGRIDIENT");
+        setAllIngridients((prev) => [...prev, mealIngridient]);
```

Comment:
Nit. Try to avoid running this function inside loop. It can be heavy.
Instead, you can can create an array, push to it and in the end add array to all ingredients.

It might be also to consider a different data structure for all ingredients. Such as `Set` or hash map with more efficient checks.